### PR TITLE
DOP-1643: Store static_assets hashes along with a reference string

### PR DESCRIPTION
**Pull request by i80and**

### CODE REVIEW

File: snooty/types.py [hunk 1]
```diff
@@ -45,6 +45,9 @@ class SnootyError(Exception):
 
 @dataclass
 class StaticAsset:
+    # "key" must *exactly* match an identifier with which this asset is referred to in source text.
+    key: str
+
```
Comment (sophstad) [inline snooty/types.py:50]: [nit] Accidental newline?

**Review by sophstad — COMMENTED** [review]
Wondering if you think it makes sense to remove the `filename` field from a document in the `assets` collection (i.e. [this](https://github.com/mongodb/snooty-parser/blob/master/snooty/main.py#L223)). A few reasons in favor:

- We explicitly support using multiple filenames to refer to the same image, so this field is not guaranteed to be correct
- The front end will no longer use this field when writing image files

File: snooty/types.py (hunk 1 above)
Comment (i80and) [reply snooty/types.py:50]: Intentional -- there's not really a good way to clearly document instance variables, so my convention is to have a newline after each documented instance variable

**Comment by i80and** [comment]
> Wondering if you think it makes sense to remove the `filename` field from documents in the `assets` collection (i.e. [this](https://github.com/mongodb/snooty-parser/blob/master/snooty/main.py#L223)). A few reasons in favor:
> 
>     * We explicitly support using multiple filenames to refer to the same image, so this field is not guaranteed to be correct
> 
>     * The front end will no longer use this field when writing image files
> 
> 
> We could also consider whether it makes sense to drop the `assets` collection entirely if this change is made.

The original intent of the `filename` field was purely debugging documentation. I'm not super involved with the ops/debugging side of things, so if losing that metadata is fine, I'm okay with removing the filename field!

**Review by sophstad — APPROVED** [review]

**Pull request merged**